src/bin/cli/mod.rs

```diff
@@ -682,3 +697,96 @@ fn explicit_args(matches: &ArgMatches) -> Vec<&Id> {
         })
         .collect()
 }
+
+/// A helper function to check if the target name provided by the user matches the target name in solang.toml
+///
+/// Returns false if the target names do not match
+/// Returns true if the target names match or if solang.toml does not exist in the current directory
+///
+/// ## Errors:
+///
+/// 1- Returns an error if the solang.toml file does not exist in the current directory.
+/// 2- Returns an error if the solang.toml file exists but it is not valid TOML.
+/// 3- Returns an error if the solang.toml file exists but it does not contain a target name.
```

Comment:
I don't think this is valid markdown syntax. Also the comment is wrong and confusing. It should mention that this has the side effect of reading `./solang.toml` if the config file content arg is `None`. And that it will return errors if the provided config file String is invalid.